src/resource.ts

```diff
@@ -508,3 +509,34 @@ function optionsToRequestInit(method: string, options?: GetRequestOptions | Post
 
 }
 
+class ActiveRefreshes<T = any> {
+  private readonly refreshBySerializedRequest: Map<string, Promise<State<T>>> = new Map<string, Promise<State<T>>>();
+
+  put(activeRefresh: Promise<State<T>>, uri: string, options?: GetRequestOptions): void {
+    this.refreshBySerializedRequest.set(ActiveRefreshes.serializeRequest(uri, options), activeRefresh);
+  }
+
+  get(uri: string, options?: GetRequestOptions): Promise<State<T>> | undefined {
+    return this.refreshBySerializedRequest.get(ActiveRefreshes.serializeRequest(uri, options));
+  }
+
+  has(uri: string, options?: GetRequestOptions): boolean {
+    return !!this.get(uri, options);
+  }
+
+  remove(uri: string, options?: GetRequestOptions): boolean {
+    return this.refreshBySerializedRequest.delete(ActiveRefreshes.serializeRequest(uri, options));
+  }
+
+  private static serializeRequest(uri: string, options?: GetRequestOptions): string {
+    if (!options) {
+      return objectHash(uri);
+    }
+    const sortedHeaders: any = {};
+    new Headers(options.getContentHeaders?.() || options.headers)
+      .forEach((value, key) => {
+        sortedHeaders[key] = [value.split(', ').sort()];
```

Comment:
If a key presents itself twice (I think it is possible), you are going to loose values.

Reply:
By doing `new Headers.foreach((value, key) => {})`, the duplicated keys will be flatten in the foreach value.
Working on a test to assert this.

Reply:
Test added